frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -94,12 +98,36 @@ export const useUserSettings = <T>(
   }, [cfLoadError, cfLoaded, fallbackLocalStorage]);
 
   React.useEffect(() => {
-    if (cfData && cfLoaded && settings !== undefined) {
-      updateConfigMap(cfData, keyRef.current, seralizeData(settings));
+    if (sync && !isRequestPending.current) {
```

Comment:
The changes in `useUserSettings` will conflict with @sahil143's PR. I suggest you base this PR on his work.

Reply:
these changes have been copied from Sahil's https://github.com/openshift/console/pull/7298 PR, which he has closed yesterday. I have updated my PR description that it depends on https://github.com/openshift/console/pull/7291.